src/js/controller/controller.js

```diff
@@ -298,57 +293,61 @@ Object.assign(Controller.prototype, {
         };
 
         function _load(item, feedData) {
-            if (_model.get('state') === STATE_ERROR) {
-                _model.set('state', STATE_IDLE);
-            }
 
             _this.trigger('destroyPlugin', {});
             _stop(true);
 
             checkAutoStartCancelable.cancel();
             checkAutoStartCancelable = cancelable(_checkAutoStart);
-            _model.once('itemReady', checkAutoStartCancelable.async);
 
             switch (typeof item) {
                 case 'string':
-                    _loadPlaylist(item);
-                    break;
-                case 'object': {
-                    try {
-                        const playlist = Playlist(item);
-                        setPlaylist(_model, playlist, feedData);
-                    } catch (error) {
+                    _loadPlaylist(item).then(data => {
+                        return _updatePlaylist(data.playlist, data);
+                    }).then(checkAutoStartCancelable.async).catch(error => {
                         _this.triggerError({
                             message: `Error loading playlist: ${error.message}`
                         });
-                        return;
-                    }
-                    loadProvidersForPlaylist(_model).catch(error => {
-                        _this.triggerError({
-                            message: `Could not play video: ${error.message}`
-                        });
                     });
-                    _setItem(0);
                     break;
-                }
+                case 'object':
+                    _updatePlaylist(item, feedData).then(checkAutoStartCancelable.async);
+                    break;
                 case 'number':
-                    _setItem(item);
+                    _setItem(item).then(checkAutoStartCancelable.async);
                     break;
                 default:
                     break;
             }
         }
 
+        function _updatePlaylist(data, feedData) {
+            try {
+                const playlist = Playlist(data);
+                setPlaylist(_model, playlist, feedData);
+            } catch (error) {
+                _this.triggerError({
+                    message: `Playlist error: ${error.message}`
+                });
+                return;
```

Comment:
If we get an error, code depending on this method returning a promise will throw exceptions

Reply:
Can you provide an example?

Reply:
Got it!

Reply:
95e60bf3af4a71bf65d262f7b2e5b6e174822129